File: task-03/sub-task03/Elixir/pattern.exs
defmodule Pattern do
  def print(rows) do
    for i <- 0..rows - 1 do
      for j <- 0..rows - i - 1 do
        IO.write(" ")
      end
      for k <- 0..2 * i do
        IO.write("*")
      end
      IO.puts("")
    end

    for i <- rows - 2 downto 0 do
      for j <- 0..rows - i - 1 do
        IO.write(" ")
      end
      for k <- 0..2 * i do
        IO.write("*")
      end
      IO.puts("")
    end
  end
end

IO.puts("Enter the number of rows:")
rows = String.to_integer(IO.gets())
Pattern.print(rows)

File: task-03/sub-task03/Ruby/pattern.rb
def print_pattern(rows)
    for i in 0..rows-1
      for j in 0..rows-i-1
        print " "
      end
      for k in 0..2*i
        print "*"
      end
      puts
    end
  
    for i in (rows-2).downto(0)
      for j in 0..rows-i-1
        print " "
      end
      for k in 0..2*i
        print "*"
      end
      puts
    end
  end
  
  rows = 3
  print_pattern(rows)